### DISCUSSION TITLE
How to test two services communicating with pytest.fixture

### DISCUSSION THREAD

**Discussion by frucht-salat** [fastapi/fastapi#12260] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import re
import json
import pytest
import requests
import responses
from fastapi.testclient import TestClient
from fastapi import FastAPI, Request
from contextlib import asynccontextmanager
from functools import partial


@asynccontextmanager
async def lifespan(app: FastAPI):
    yield


app = FastAPI(lifespan=lifespan)


@app.post("/test/{id}")
async def run_id(request: Request, payload: dict, id: str):
    if int(id) < 3:
        requests.post(
            f'http://www.example.com/test/{int(id) + 1}',
            json=payload,
            headers={"Content-Type": "application/json"}
        )


def mock_func(request, client):
    client.post(
        request.url,
        json=json.loads(request.body),
        headers={"Content-Type": "application/json"}
    )
    return 200, {}, request.body


@pytest.fixture(scope='function')
def client():
    with TestClient(app) as client:
        yield client


@pytest.fixture
def mock_api(client):
    with responses.RequestsMock() as rsps:
        rsps.add_callback(
            responses.POST,
            re.compile('http://www.example.com/test/[a-zA-Z0-9_-]*'),
            callback=partial(mock_func, client=client),
            content_type="application/json",
        )
        yield rsps


def test_meta_task(mock_api):
    with TestClient(app) as client:
        client.post('/test/1', json={"test": "data"})
```

### Description

My mock_func is actually another service doing something, then communicating back to the testclient. I want to simulate the communication between both services, but get the error:

```
self = <anyio._backends._asyncio.BlockingPortal object at 0x7f8e419965f0>

def _check_running(self) -> None:
        if self._event_loop_thread_id is None:
            raise RuntimeError("This portal is not running")
        if self._event_loop_thread_id == threading.get_ident():
>           raise RuntimeError(
                "This method cannot be called from the event loop thread"
            )
E           RuntimeError: This method cannot be called from the event loop thread
```
This probably happens so that its not possible to have deadlocks in the code, but if I need that behavior to work, is there a way to circumvent this RuntimeError?

Thank you!

### Operating System

Linux, Windows

### Operating System Details

WSL2 - Ubuntu 22.04 on Windows

### FastAPI Version

0.101.1

### Pydantic Version

1.10.17

### Python Version

3.10.12

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
